src/rws_interface.cpp

```diff
@@ -117,6 +117,111 @@ std::string RWSInterface::getIOSignal(const std::string& iosignal)
+  {
+    RWSClient::RWSResult rws_result = rws_client_.getMechanicalUnitStaticInfo(mechunit);
+
+    if(rws_result.success)
+    {
+      p_static_info->task_name = xmlFindTextContent(rws_result.p_xml_document, XMLAttribute("class", "task-name"));
+      p_static_info->is_integrated_unit = xmlFindTextContent(rws_result.p_xml_document,
+                                                             XMLAttribute("class", "is-integrated-unit"));
+      p_static_info->has_integrated_unit = xmlFindTextContent(rws_result.p_xml_document,
+                                                              XMLAttribute("class", "has-integrated-unit"));
+
+      std::string type = xmlFindTextContent(rws_result.p_xml_document, XMLAttribute("class", "type"));
+      if(type == "None")
+      {
+        p_static_info->type = NONE;
+      }
+      else if(type == "TCPRobot")
+      {
+        p_static_info->type = TCP_ROBOT;
+      }
+      else if(type == "Robot")
+      {
+        p_static_info->type = ROBOT;
+      }
+      else if(type == "Single")
+      {
+        p_static_info->type = SINGLE;
+      }
+      else
+      {
+        p_static_info->type = UNDEFINED;
+      }
+
+      std::stringstream axes(xmlFindTextContent(rws_result.p_xml_document, XMLAttribute("class", "axes")));
+      axes >> p_static_info->axes;
+
+      std::stringstream axes_total(xmlFindTextContent(rws_result.p_xml_document, XMLAttribute("class", "axes-total")));
+      axes_total >> p_static_info->axes_total;
+
+      result = true;
+    }
+  }
+
+  return result;
+}
+
+bool RWSInterface::getMechanicalUnitDynamicInfo(const std::string& mechunit, MechanicalUnitDynamicInfo* p_dynamic_info)
+{
+  bool result = false;
+
+  if (p_dynamic_info)
+  {
+    RWSClient::RWSResult rws_result = rws_client_.getMechanicalUnitDynamicInfo(mechunit);
+
+    if(rws_result.success)
+    {
+      p_dynamic_info->tool_name = xmlFindTextContent(rws_result.p_xml_document, XMLAttribute("class", "tool-name"));
+      p_dynamic_info->wobj_name = xmlFindTextContent(rws_result.p_xml_document, XMLAttribute("class", "wobj-name"));
+      p_dynamic_info->payload_name = xmlFindTextContent(rws_result.p_xml_document,
+                                                        XMLAttribute("class", "payload-name"));
+      p_dynamic_info->total_payload_name = xmlFindTextContent(rws_result.p_xml_document,
+                                                              XMLAttribute("class", "total-payload-name"));
+      p_dynamic_info->status = xmlFindTextContent(rws_result.p_xml_document, XMLAttribute("class", "status"));
+      p_dynamic_info->jog_mode = xmlFindTextContent(rws_result.p_xml_document, XMLAttribute("class", "jog-mode"));
+
+      std::string mode = xmlFindTextContent(rws_result.p_xml_document, XMLAttribute("class", "mode"));
+      if(mode == "Activated")
+      {
+        p_dynamic_info->mode = ACTIVATED;
+      }
+      else
+      {
+        p_dynamic_info->mode = DEACTIVATED;
+      }
```

Comment:
Perhaps on purpose, but this won't catch corrupted data, will it?

Should we perhaps explicitly compare with both valid values?

Reply:
Good idea to check only the valid values.

I can add something like an unknown mode.

Reply:
Included in b38e52c

Reply:
So if the mode is not recognised, it's set to `UNKNOWN`, correct (or actually: never changed from the initial `UNKNOWN`)?

Reply:
Correct